examples/feature_selection/plot_rfe_with_cross_validation.py

```diff
@@ -3,19 +3,23 @@
 Recursive feature elimination with cross-validation
 ===================================================
 
-A recursive feature elimination example with automatic tuning of the
+A Recursive Feature Elimination (RFE) example with automatic tuning of the
 number of features selected with cross-validation.
 
 """
 
-import numpy as np
-import matplotlib.pyplot as plt
-from sklearn.svm import SVC
-from sklearn.model_selection import StratifiedKFold
-from sklearn.feature_selection import RFECV
+# %%
+# Data generation
+# ---------------
+#
+# We build a classification task using 3 informative features. The introduction
+# of 2 additional redundant (i.e. correlated) features has the effect that the
+# selected features vary depending on the cross-validation fold. The remaining
+# features are drawn at random and discarding them should not considerably
+# affect the test scores.
+
 from sklearn.datasets import make_classification
 
-# Build a classification task using 3 informative features
 X, y = make_classification(
```

Comment:
You can try the following:

```python
X, y = make_classification(
    n_samples=500,
    n_features=15,
    n_informative=3,
    n_redundant=2,
    n_repeated=0,
    n_classes=8,
    n_clusters_per_class=1,
    class_sep=0.8,
    random_state=0,
)
```

We make the problem slightly more difficult and reduce the number of features (that reduce later the processing of the `RFECV`)